### DISCUSSION TITLE
Fn + F6 Not Working with Fn Lock?

### DISCUSSION THREAD

**Discussion by FlyingFaller** [seerge/g-helper#633] [General]
Not sure if this is intended and I am too scared to make a bug report on something I don't know the intended behavior of. 

I have the Zephyrus M16 GU603HM and discovered this wonderful tool. The F6 key has a special function that launches the Windows Snipping Tool. When Fn-Lock is enabled, all other special functions work as intended except F6 snipping tool. Snipping tool is only summoned when Fn+F6 is pressed regardless of Fn Lock. 

[log.txt](https://github.com/seerge/g-helper/files/11763915/log.txt)
I have attached my logs which show F6 is being picked up.

Windows 11 22H2 22621.1848

**Comment by seerge** [comment]
@FlyingFaller it emits a Print-Screen button, if you want windows to show snipping tool you can set that in windows

![prntscrn](https://i0.wp.com/pureinfotech.com/wp-content/uploads/2021/08/windows-11-settings-print-screen-button.jpg?resize=827%2C402&quality=78&strip=all&ssl=1)

**Reply by FlyingFaller** [reply]
Thank you for clarifying this worked. From what I can gather, when Fn-Lock is on Fn+Function_Key is always taken regardless of whether you actually hold Fn. This is extremely useful, but is there a way to do an Fn invert? i.e. You press Fn to access the normal function keys and don't to access the special keys? If not maybe consider it a feature request if it is within the realm of possibility. Already super impressed you managed to get Fn-Lock working as I know you can't go through the usual windows interfaces.

**Reply by seerge** [reply]
I'm keen to keep it as is (i.e with printscreen action) as it gives more flexibility :)

**Reply by FlyingFaller** [reply]
I have no qualms with the choice to assign it the printscreen action. I just proposed a new or alternate feature which I would find more useful and would restore more functionality than regular Fn-Lock. The idea is that pressing a function key does the advertised special action (like F6 -> printscreen or F7 -> lower brightness) and pressing Fn+Function Key does the function key (like Fn+F4 -> F4). This effectively _inverts_ the Fn key. 

This is s different from the Fn-Lock behavior which works like F6-> printscreen _and_ Fn+F6 -> printscreen. This means there is no way to use the F6 key normally without disabling Fn-Lock first. 

I would find this more useful than Fn-Lock or nothing at all because I use the special functions (like brightness) more frequently than the function keys but I still use both on a day-to-day basis.

**Reply by seerge** [reply]
Ah, i must have miss read then. Reverse fn-lock won't work due technical limitations

**Reply by FlyingFaller** [reply]
Got it. I'm guessing you just rebound the function keys to custom actions and there would be no way of actually detecting an Fn key press. Still very useful! I will close the discussion.
